src/components/panel/panel.js

```diff
@@ -806,6 +860,79 @@ MdPanelService.prototype.newPanelAnimation = function() {
 
 
 /**
+ * Creates a panel group and adds it to a tracked list of panel groups.
+ * @param groupName {string} Name of the group to create.
+ * @param config {!Object=} Specific configuration object that may contain the
+ *     following properties:
+ *
+ *   - `maxOpen` - `{number=}`: The maximum number of panels that are allowed
+ *     open within a defined panel group.
+ */
+MdPanelService.prototype.newPanelGroup = function(groupName, config) {
+  if (!this._groups.hasOwnProperty(groupName)) {
+    config = config || {};
+    var group = {
+      panels: [],
+      maxOpen: config.maxOpen > 0 ? config.maxOpen : false
+    };
+    this._groups[groupName] = group;
+    return this._groups[groupName];
+  }
+};
+
+
+/**
+ * Sets the maximum number of panels in a group that can be opened at a given
+ * time.
+ * @param {string} groupName The name of the group to configure.
+ * @param {number|boolean} maxOpen The maximum number of panels that can be
+ *     opened. If false is passed in, the maximum limit will be removed.
+ */
+MdPanelService.prototype.setGroupMaxOpen = function(groupName, maxOpen) {
+  if (this._groups[groupName]) {
+    this._groups[groupName].maxOpen = maxOpen;
+  } else {
+    throw new Error('Group does not exist yet. Call newPanelGroup().');
+  }
+};
+
+
+/**
+ * Determines if opening a panel within a group will exceed the limit of allowed
+ * open panels. If so, the first panel opened within the group will be closed.
+ * @param {string} groupName The name of the group to check.
+ * @private
+ */
+MdPanelService.prototype._checkGroupMaxOpen = function(groupName) {
+  if (groupName &&
+      this._groups[groupName] &&
+      this._groups[groupName].maxOpen > 0) {
+        var openCount = 0;
+        var firstOpenedPanel = null;
+        var firstOpenedPanelIndex = null;
+
+        angular.forEach(this._groups[groupName].panels, function(panel, index) {
+          if (panel.isOpened) {
+            openCount++;
```

Comment:
@ErinCoughlan It has to do less with the number of open panels and the determining of which panel in the group is the `firstOpenedPanel`. I could keep the opened panel count somewhere else, but I would still need to iterate in order to find the first opened panel. Is there a better way to get this setting?